src/plots/plots.js

```diff
@@ -527,6 +563,9 @@ plots.supplyDataDefaults = function(dataIn, dataOut, layout) {
         basePlotModules = layout._basePlotModules = [],
         cnt = 0;
 
+    // Make this idempotent by emptying the existing array:
+    dataOut.length = 0;
```

Comment:
But yeah, you're right. This avenue was kind of a dead end, hence the comment about the futility of just supplying all the defaults.

![](https://cdn.meme.am/instances/500x/71294786.jpg)